content/blog/2019-11-13-multiplication-strength-reduction.md

```diff
@@ -0,0 +1,54 @@
+# Strength Reduction for Multiplication
+
+## Motivation
+
+The strength reduction method is to replace expensive operations with a cheaper but equivalent one, so we can obtain a faster program. For this project, we will focus on the weak form of strength reduction. Specifically, we focus on how to replace constant multiplications with cheaper operations.
+
+Most modern processors have different latencies and throughputs for different kinds of instructions. It is sometimes possible to find instructions that are mathematically equivalent but fasters in practice. For example, on some processors, multiplication may run slower than a bitwise shift. Therefore, it is possible to replace multiplication `x * 2` with bitwise left shift operation `x << 1` for better performance.
```

Comment:
- fasters -> faster
- I'd say that that's true on _most_ processors.
- replace _a_ multiplication … with _a_ bitwise